chiselFrontend/src/main/scala/chisel3/core/Data.scala

```diff
@@ -54,6 +54,10 @@ object SpecifiedDirection {
 sealed abstract class ActualDirection
 
 object ActualDirection {
+  /** The object does not exist / is empty and hence has no direction
+    */
+  case object Empty extends ActualDirection
```

Comment:
Is this an API we want to expose or is it kind of an implementation detail?

Reply:
Fair point, it does add unnecessary complexity - let me see if I can refactor this out.